smarts/core/bubble_manager.py

```diff
@@ -108,6 +114,36 @@ def is_admissible(self, vehicle_id: str, other_vehicle_ids: Sequence[int]):
 
         return True
 
+    @property
+    def is_travelling(self):
+        return self._bubble.pinned_actor_id is not None
+
+    # TODO: Make this more efficient
+    def move_to_pinned_vehicle(self, vehicle: Vehicle):
+        x, y, _ = vehicle.position
+
+        def _transform(geom):
+            centroid = geom.centroid
+
+            # Bring back to origin
+            geom = translate(geom, xoff=-centroid.x, yoff=-centroid.y)
+            geom = rotate(geom, -self._bubble_heading, "centroid", use_radians=True)
+
+            # Now apply new transformation in "vehicle coordinate space"
+            geom = translate(
+                geom,
+                xoff=self._bubble.pinned_offset[0],
+                yoff=self._bubble.pinned_offset[1],
```

Comment:
consider exposing a rotational pinned offset as well?

Reply:
Could be valuable!